front/src/applications/stdcm/components/StdcmLayersModal.tsx

```diff
@@ -0,0 +1,157 @@
+import signalsIcon from 'assets/pictures/layersicons/layer_signal.svg';
+import pslsIcon from 'assets/pictures/layersicons/layer_tivs.svg';
+import OPsSVGFile from 'assets/pictures/layersicons/ops.svg';
+import switchesIcon from 'assets/pictures/layersicons/switches.svg';
+import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import { Modal } from 'common/BootstrapSNCF/ModalSNCF';
+import SwitchSNCF from 'common/BootstrapSNCF/SwitchSNCF/SwitchSNCF';
+import { Icon2SVG } from 'common/Map/Settings/MapSettingsLayers';
+import { useInfraID } from 'common/osrdContext';
+import type { LayersSettings } from 'reducers/map';
+import { stdcmConfInitialState, updateStdcmLayers } from 'reducers/osrdconf/stdcmConf';
+import { getStdcmSelectedLayers } from 'reducers/osrdconf/stdcmConf/selectors';
+import { useAppDispatch } from 'store';
+
+const LAYERS = [
+  { layer: 'track_sections', icon: <MdSpeed className="mx-2" /> },
+  { layer: 'signals', icon: signalsIcon },
+  { layer: 'buffer_stops', icon: bufferStopIcon },
+  { layer: 'detectors', icon: detectorsIcon },
+  { layer: 'switches', icon: switchesIcon },
+  { layer: 'sncf_psl', icon: pslsIcon },
+  { layer: 'electrifications', icon: <GiElectric className="mx-2" style={{ width: '20px' }} /> },
+  { layer: 'neutral_sections', icon: <GiUnplugged className="mx-2" style={{ width: '20px' }} /> },
+  {
+    layer: 'platforms',
+    icon: <TbRectangleVerticalFilled className="mx-2" style={{ width: '20px' }} />,
+  },
+  {
+    layer: 'operational_points',
+    icon: <Icon2SVG file={OPsSVGFile} style={{ width: '20px' }} className="mx-2" />,
+  },
+  { layer: 'speed_limits', icon: <MdSpeed style={{ width: '20px' }} className="mx-2" /> },
+];
+
+type StdcmLayersModalProps = {
+  initialLayers: LayersSettings | undefined;
+  onChange: (args: { newLayers: LayersSettings }) => void;
+};
+
+const StdcmLayersModal = ({ initialLayers, onChange }: StdcmLayersModalProps) => {
+  const dispatch = useAppDispatch();
+  const { t } = useTranslation();
+  const [selectedLayers, setSelectedLayers] = useState<LayersSettings>(
+    initialLayers || stdcmConfInitialState.layersSettings
+  );
+  const layersSettings = useSelector(getStdcmSelectedLayers);
+
+  const infraID = useInfraID();
+
+  const { data: speedLimitTagsByInfraId } =
+    osrdEditoastApi.endpoints.getInfraByInfraIdSpeedLimitTags.useQuery(
+      { infraId: infraID! },
+      { skip: !infraID }
+    );
+  const { data: speedLimitTags } = osrdEditoastApi.endpoints.getSpeedLimitTags.useQuery();
+
+  const allSpeedLimitTags = compact(concat(speedLimitTags, speedLimitTagsByInfraId));
+  const allSpeedLimitTagsOrdered = useMemo(() => allSpeedLimitTags.sort(), [allSpeedLimitTags]);
+
+  const DEFAULT_SPEED_LIMIT_TAG = useMemo(() => t('Editor.layers-modal.noSpeedLimitByTag'), [t]);
+
+  const speedLimitOptions = useMemo(
+    () => uniq([DEFAULT_SPEED_LIMIT_TAG, ...allSpeedLimitTagsOrdered]),
+    [allSpeedLimitTagsOrdered, selectedLayers]
+  );
```

Comment:
Actually, you can merge allSpeedLimitTags, allSpeedLimitTagsOrdered and speedLimitOptions, since you use allSpeedLimitTags and allSpeedLimitTagsOrdered to compute speedLimitOptions and they are not used anywhere else.
```suggestion
  const DEFAULT_SPEED_LIMIT_TAG = useMemo(() => t('Editor.layers-modal.noSpeedLimitByTag'), [t]);
  
  const speedLimitOptions = useMemo(() => {
    const allSpeedLimitTags = uniq(compact(concat(speedLimitTags, speedLimitTagsByInfraId))).sort();
    return [t('Editor.layers-modal.noSpeedLimitByTag'), ...allSpeedLimitTags];
  }, [t, speedLimitTags, speedLimitTagsByInfraId]);
```